### PR TITLE
Updates responsive embeds with new class names and CSS variables

### LINKED ISSUE #31531: Add responsive embed class with ratio by variable
It would be handy if there was support for a variable ratio.

For example, 

```
<style>
    .embed-responsive-by-ratio::before {
        padding-top: 56.25%; /* 16:9 fallback when var() not supported */
        padding-top: var(--embed-responsive-by-ratio);
    }
</style>

<div class="embed-responsive embed-responsive-by-ratio" style="--embed-responsive-by-ratio: 17.8%;"></div>
```

When templating a known width and height it is simpler to calculate the padding than determine which of your classes is a closer fit.

Didn't see this in the documentation but it is pretty simple and working for me - would probably be generally useful

### PR DESCRIPTION
**Pull request by mdo**
- Renames .embed-responsive-XbyY classes to .embed-responsive-XxY
- Simplifies the Sass map to just be key and value, ratio and percentage
- Builds .embed-responsive-* modifiers with CSS variables
- Updates docs to show power of CSS variables
- Add notes to the Migration guide

Fixes #31531.

---

Screen Shot 2020-09-21 at 2 15 25 PM

Screen Shot 2020-09-21 at 2 15 30 PM

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -378,22 +378,10 @@ $transition-collapse:         height .35s ease !default;
...
-    y: 3
-  ),
-  "1by1": (
-    x: 1,
-    y: 1
-  )
+  "1x1": 100%,
+  "4x3": 75%,
+  "16x9": 56.25%,
+  "21x9": 42.857142%
```
[the remark this answers is not in the document]
Comment (mdo) [reply scss/_variables.scss:384]: `calc(21/9%)` doesn't work. We'd have to do something like `calc(calc(9/21) * 100%)`, which doesn't feel more readable to me.

[the remark this answers is not in the document]
Comment (MartijnCuppens) [reply scss/_variables.scss:384]: `calc(9 / 21 * 100%)` or `calc(900% / 21)` should do the trick.

File: site/content/docs/5.0/helpers/embed.md [hunk 2]
```diff
@@ -26,28 +28,35 @@ Wrap any embed like an `<iframe>` in a parent element with `.embed-responsive` a
...
-<div class="embed-responsive embed-responsive-16by9">
-  <iframe class="embed-responsive-item" src="..."></iframe>
+<div class="embed-responsive embed-responsive-4x3" style="width: 10rem;">
+  <div class="embed-responsive-item">4x3</div>
 </div>
-
-<!-- 4:3 aspect ratio -->
-<div class="embed-responsive embed-responsive-4by3">
-  <iframe class="embed-responsive-item" src="..."></iframe>
+<div class="embed-responsive embed-responsive-16x9" style="width: 10rem;">
```
Comment (MartijnCuppens) [inline site/content/docs/5.0/helpers/embed.md:38]: I would move the `style` to the docs css, this might get copy-pasted.

**Comment by MartijnCuppens** [comment]
Thoughts about renaming `embed-responsive` to `ratio`?

File: scss/_variables.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (ffoodd) [reply scss/_variables.scss:384]: Agreed, I prefer self-explanatory values even if they can be simpler another way. The percentage result looks like a magical number but isn't.

**Comment by mdo** [comment]
Think that'd be fine @MartijnCuppens. We'd change the `.responsive-embed-item` too, and keep the nested selectors, right?

**Comment by mdo** [comment]
Done, and done, and done! Latest commits rename it to `.ratio` and move inline styles to the docs CSS. Also added an example of using breakpoints to change the CSS variable for responsive ratios.

**Review by ffoodd — APPROVED** [review]
LGTM!

**Comment by MartijnCuppens** [comment]
Very nice

One last thought:

We could simplify this to just `> *`:
https://github.com/twbs/bootstrap/blob/f407d32134c8e6b09644c12fa51af56e5a135ce8/scss/helpers/_ratio.scss#L13-L17
It feels a bit inconsistent that we would require `.ratio-item` on a `<div>` or an ``, but not on an `<iframe>` or a `<video>`

**Comment by mdo** [comment]
Yup, good idea—just pushed some changes.Done!

**Review by MartijnCuppens — APPROVED** [review]
:shipit:

**Pull request merged**